### PR TITLE
Bug fixes (payouts, masternode coins)

### PR DESCRIPTION
**Pull request by kidcosmic**
- Fix masternode payment calculations for old masternode blocktemplate, ie
- Fix block submit crash on masternode coins with no votes
- Fix submit block tracking/payouts, pending/confirmed blocks working, payout's working.
  
Tested on Magnet, Pepe, Monoeci (Monaco)

### CODE REVIEW

**Comment by foxer666** [comment]
Nice! Thank You very much!

File: lib/pool.js [hunk 1]
```diff
@@ -319,11 +317,13 @@ var pool = module.exports = function pool(options, authorizeFn){
                 emitShare();
             else{
                 SubmitBlock(blockHex, function(){
-                    CheckBlockAccepted(shareData.blockHash, function(isAccepted, tx){
+                    shareData.blockHash = util.reverseBuffer(new Buffer(blockHex.toString('hex').substr(8, 64), 'hex')).toString('hex');
```
[the remark this answers is not in the document]
Comment (kidcosmic) [reply lib/pool.js]: You're correct. We should be using the calculated blockhash from the submitblock header

[the remark this answers is not in the document]
Comment (kidcosmic) [reply lib/pool.js]: `blockHash = hashAlgo(hashAlgo(blockHeader))`

Link: https://en.bitcoin.it/wiki/Block_hashing_algorithm
> Bitcoin uses: SHA256(SHA256(Block_Header)) but you have to be careful about byte-order.

[the remark this answers is not in the document]
Comment (devnull-ed) [reply lib/pool.js]: Yeah, I dealt with it. But, how to find out which hashing algorithm is used to hash header? For example, in ZCoin, SHA256 is used, so thats why it wasnt working. I added additional key in coin config for this, but still.

[the remark this answers is not in the document]
Comment (kidcosmic) [reply lib/pool.js]: So it looks like we should already have a correct blockHash.
The block info we're getting here is calculated in the JobManager at:

https://github.com/foxer666/node-stratum-pool/blob/bfe370a1568d745bcfe18bbb48cf8a414c07da9f/lib/jobManager.js#L247

So if we just comment this line out, we should be good.
Try it out and let me know.

**Pull request merged**